sdk/src/client/bindings/python/iota_client/types/address.py

```diff
@@ -0,0 +1,67 @@
+from enum import Enum
+
+class AddressType(Enum):
+    ED25519 = 0
+    ALIAS = 8
+    NFT = 16
+
+class Address():
+    def __init__(self, type, address_or_id):
+        """Initialize an Address
+        
+        Parameters
+        ----------
+        type : AddressType
+            The type of the Address
+        address_or_id : string
+            The address to use. Can either be an hex encoded ED25519 address or NFT/Alias id
+        """
+        self.type = type
+        self.address_or_id = address_or_id
```

Comment:
This is pretty awkward. Is there no better way?

Reply:
No idea, @Dr-Electron do you know more?

Reply:
Not that I know of. No. That's just Python :trollface:

Reply:
Well, the other constructors use separate fields to define the variants. This one uses a single field.

Reply:
It's just a single field, it's called like this because alias id == alias address and nft id == nft address, so both work

Reply:
Right, but wouldn't it be better to use different field names?

Reply:
Multiple fields which will be the same field at the end?

Reply:
But they aren't the same field if it's an address OR an id.

Reply:
Currently it's `config['aliasId'] = config.pop('address_or_id')`

Splitting to `alias_id` and `alias_address` would end up in `config['aliasId'] = config.pop('alias_id')` and `config['aliasId'] = config.pop('alias_address')` depending on what was provided

Reply:
I'm confused. This is handled just find by the other types. Like Unlock Condition for instance:

```py
        if 'type' in config:
            config['type'] = config['type'].value

        if 'address' in config:
            config['address'] = config['address'].as_dict()

        if 'returnAddress' in config:
            config['returnAddress'] = config['returnAddress'].as_dict()

        if 'amount' in config:
            config['amount'] = str(config['amount'])
```

Not all fields are on each variant. The same would apply to this, no?

Reply:
Yeah but in this case address and ID have the same representation, they really are interchangeable

Reply:
There the values are different, `address` and `returnAddress` will not be used for the same thing

You propose parameters like here?
`pub_key_hash=None, alias_address_or_id=None, nft_address_or_id=None`
Or maybe even also with address and id splitted
`pub_key_hash=None, alias_address=None, alias_id=None, nft_address=None, nft_id=None`

So just more parameters which will not have a different outcome?

Reply:
Alright, fair enough.